### PR TITLE
EC 23U PCB

### PR DESCRIPTION
**Pull request by Cipulot**
Addition of the EC 23U PCB firmware. Open source universal PCB for the Realforce 23U numpad.
## Description

The EC 23U PCB is a replacement PCB for the Realforce 23U numpad. Apart from the OEM layout, it enables other layout options and boosts NumLock indicator.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/cipulot/ec_23u/keymaps/via/via_apc.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+// Declaring enums for VIA config menu
+enum via_apc_enums {
+    // clang-format off
+    id_apc_actuation_threshold = 1,
+    id_apc_release_threshold = 2
+    // clang-format on
+};
+
+// Initializing persistent memory configuration: default values are declared and stored in PMEM
+void eeconfig_init_kb(void) {
```
Comment (zvecr) [inline keyboards/cipulot/ec_23u/keymaps/via/via_apc.c:45]: Cant have `_kb` functions at the keymap level

File: keyboards/cipulot/ec_23u/keymaps/via/via_apc.c [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    // Default values
+    apc.actuation_threshold = DEFAULT_ACTUATION_LEVEL;
+    apc.release_threshold   = DEFAULT_RELEASE_LEVEL;
+    // Write default value to EEPROM now
+    eeconfig_update_kb_datablock(&apc);
```
Comment (zvecr) [inline keyboards/cipulot/ec_23u/keymaps/via/via_apc.c:50]: Nor should it be using the kb eeprom block

File: keyboards/cipulot/ec_23u/info.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
+{
+    "manufacturer": "Cipulot",
+    "keyboard_name": "EC 23U",
+    "maintainer": "Cipulot",
+    "board": "GENERIC_STM32_F401XC",
```
Comment (zvecr) [inline keyboards/cipulot/ec_23u/info.json:5]: This is the default?
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/cipulot/ec_23u/info.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
+    },
+    "keyboard_folder": "cipulot/ec_23u",
```
Comment (zvecr) [inline keyboards/cipulot/ec_23u/info.json:22]: `keyboard_folder` should never be set
```suggestion
```

**Review by zvecr — COMMENTED** [review]
These seem to be consistent across all your open PR, so you should also fix those as well.

File: keyboards/cipulot/ec_23u/info.json (hunk 3 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/info.json]: Done 👍

File: keyboards/cipulot/ec_23u/info.json (hunk 4 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/info.json]: Done 👍

File: keyboards/cipulot/ec_23u/keymaps/via/via_apc.c (hunk 1 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/keymaps/via/via_apc.c]: Changes tested and applied.

File: keyboards/cipulot/ec_23u/keymaps/via/via_apc.c (hunk 2 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/keymaps/via/via_apc.c]: As before.

**Comment by Cipulot** [comment]
Applied the discussed changes to this and all the other PRs.

**Comment by Cipulot** [comment]
@drashna changes applied to this PR too, following the other you pointed out 👍 .

**Comment by Cipulot** [comment]
@zvecr friendly ping for review of this and possibly the other EC series PRs waiting for merge 😄

**Comment by keyboard-magpie** [comment]
The last merges for cipulot's boards are all hardware verified, merged under the new rules (1 reviewer approval with all conditions met).

**Pull request merged**